### PR TITLE
Bigcommerce connector

### PR DESCRIPTION
**Pull request by Revathy-Ravichandran**
Added bigcommerce connector

### CODE REVIEW

File: bigcommerce.rb [hunk 1]
```diff
@@ -0,0 +1,1728 @@
+{
+  title: 'BigCommerce',
+  connection: {
+    fields: [
+      {
+        name: 'access_token',
+        control_type: 'password',
+        optional: false,
+        hint: 'Access token can be
```
Comment (mahesh-railsdata) [inline bigcommerce.rb]: Change the hint in proper way

File: bigcommerce.rb [hunk 2]
```diff
@@ -0,0 +1,1728 @@
...
+      },
+      {
+        name: 'store_hash',
+        optional: false,
+        hint: 'eg: https://api.bigcommerce.com/stores/<storehash>/v3'
+      }
+    ],
+    authorization: {
+      type: 'api_key',
+      credentials: lambda do |connection|
```
Comment (mahesh-railsdata) [inline bigcommerce.rb]: Please use apply block.

File: bigcommerce.rb [hunk 3]
```diff
@@ -0,0 +1,1728 @@
...
+        headers("X-Auth-Token": "#{connection['access_token']}")
+      end
+    },
+    base_uri: lambda do |connection|
+      "https://api.bigcommerce.com/stores/#{connection['store_hash']}/"
+    end
+  },
+  test: lambda do |connection|
+    get("v3/customers")
```
Comment (mahesh-railsdata) [inline bigcommerce.rb]: Use single quotes

File: bigcommerce.rb [hunk 4]
```diff
@@ -0,0 +1,1728 @@
...
+        page = closure['page'] || 1
+        date_created = (closure['date_created'] || input['since'] || 1.hour.ago).to_time.utc.iso8601
+        response = get('v3/customers').
+          params( page: page,
+            limit: limit,
+            'date_created:min': date_created,
+            sort: 'date_created:asc')
+        records = response['data']
+        closure = if(has_more = response&.size >= limit)
+        { 'page': response['meta']['pagination']['current_page'] + 1 }
```
Comment (mahesh-railsdata) [inline bigcommerce.rb]: Change the response to records

File: bigcommerce.rb [hunk 5]
```diff
@@ -0,0 +1,1728 @@
...
+      end,
+      sample_output: lambda do |_connection, input|
+        {
+          "created_at": 1608997218,
+          "store_id": "1001565393",
+          "producer": "stores/rqiqa1qjms",
+          "scope": "store/#{input['object']}/updated",
+          "hash": "025b76076a6cf5f49407c37c19cf53648f8784e2",
+          "data": {
+            "type": "#{input['object']}",
```
Comment (mahesh-railsdata) [inline bigcommerce.rb]: Use it directly

File: bigcommerce.rb (hunk 1 above)
Comment (Revathy-Ravichandran) [reply bigcommerce.rb]: resolved

File: bigcommerce.rb (hunk 2 above)
Comment (Revathy-Ravichandran) [reply bigcommerce.rb]: I've used apply block.

File: bigcommerce.rb (hunk 3 above)
Comment (Revathy-Ravichandran) [reply bigcommerce.rb]: Fixed

File: bigcommerce.rb (hunk 4 above)
Comment (Revathy-Ravichandran) [reply bigcommerce.rb]: Fixed

File: bigcommerce.rb (hunk 5 above)
Comment (Revathy-Ravichandran) [reply bigcommerce.rb]: Fixed

**Pull request closed without merging**